string qt_req
---
string qt_res